### PR TITLE
added a custom DB engine for V2 to add search path

### PR DESCRIPTION
**Pull request by muhammad-ali-e**
## What

- Implemented a custom Django database engine.
- Integrated support for setting the search_path for multi-tenant V2 (which may operate in a schema other than the default public schema).

## Why

- This change ensures compatibility between Django and connection poolers like PgBouncer by setting the appropriate schema search_path for each connection.

## How

- Developed a custom database engine that sets the search_path automatically for each new or reused connection.

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- No

## Database Migrations

- No

## Env Config

- 

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions

-

## Notes on Testing

-

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: backend/backend/custom_db/base.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        This ensures that the database queries will look in the specified schema.
+
+        Args:
+            connection: The database connection for which to set the search path.
+        """
+        conn_id = id(connection)
+        original_autocommit = connection.autocommit
+        try:
+            connection.autocommit = True
```
Comment (johnyrahul) [inline backend/backend/custom_db/base.py:41]: will this override the `ATOMIC_REQUESTS` of Django framework? @muhammad-ali-e

Comment (muhammad-ali-e) [reply backend/backend/custom_db/base.py:41]: We reverted this on the line  `connection.autocommit = original_autocommit`

**Review by muhammad-ali-e — COMMENTED** [review]
Addressed

File: backend/backend/custom_db/base.py (hunk 1 above)
Comment (johnyrahul) [reply backend/backend/custom_db/base.py:41]: Ah ok . Didn't noticed. Thanks

**Review by johnyrahul — APPROVED** [review]

**Review by chandrasekharan-zipstack — APPROVED** [review]

File: backend/backend/settings/base.py [hunk 2]
```diff
@@ -396,10 +396,6 @@ def get_required_setting(
             "PASSWORD": f"{DB_PASSWORD}",
             "PORT": f"{DB_PORT}",
             "ATOMIC_REQUESTS": ATOMIC_REQUESTS,
-            "OPTIONS": {
-                "options": f"-c search_path={DB_SCHEMA}",
-                "application_name": os.environ.get("APPLICATION_NAME", ""),
```
Comment (chandrasekharan-zipstack) [inline backend/backend/settings/base.py:401]: @muhammad-ali-e why was this removed? @Deepak-Kesavan had recently added this to know which service made the request

Comment (muhammad-ali-e) [reply backend/backend/settings/base.py:401]: @chandrasekharan-zipstack We are not using the options in pgbouncer initial conention.
@Deepak-Kesavan Is it needed?.
